spectrum-offchain-cardano/src/data/pool.rs

```diff
@@ -495,15 +539,8 @@ where
         output_index: 1,
         action: ClassicalOrderAction::Apply,
     };
-    let order_validator = order.get_validator(&ctx);
-    let order_script = PartialPlutusWitness::new(
-        PlutusScriptWitness::Ref(order_validator.hash),
-        order_redeemer.to_plutus_data(),
-    );
-    let order_in = SingleInputBuilder::new(order_ref.into(), order_utxo.clone())
-        .plutus_script_inline_datum(order_script, Vec::new().into())
-        .unwrap();
-    let (next_pool, user_out) = match pool.clone().apply_order(order.clone()) {
+
+    let (next_pool, operation_result_blueprint) = match pool.clone().apply_order(order.clone(), ctx.clone()) {
```

Comment:
pool is Copy, order too, as well as ctx. Why cloning everything?

Reply:
Pool have `Copy` instance, but order and ctx no